src/App.tsx

```diff
@@ -1,26 +1,168 @@
 /* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+import React, { useEffect, useState, useMemo } from 'react';
+import { Todo } from './types/Todo';
+import {
+  USER_ID,
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { ErrorStatus } from './types/ErrorStatus';
+import { Status } from './types/Status';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoList } from './components/TodoList';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [error, setError] = useState<ErrorStatus>(ErrorStatus.Empty);
+  const [filter, setFilter] = useState<Status>(Status.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodoList)
+      .catch(() => {
+        setError(ErrorStatus.UnableToLoad);
+      });
+  }, []);
+
+  const todosActiveNum = useMemo(
+    () => todoList.filter(todo => !todo.completed).length,
+    [todoList],
+  );
+
+  const todosCompletedNum = useMemo(
+    () => todoList.filter(todo => todo.completed).length,
+    [todoList],
+  );
+
+  const allTodosCompleted = useMemo(
+    () => todoList.length === todosCompletedNum,
+    [todoList],
+  );
```

Comment:
React Hook useMemo has a missing dependency: 'todosCompletedNum'. Either include it or remove the dependency array

Actually you don't need to use `useMemo` here as it's a primitive value